damnit/cli.py

```diff
@@ -185,15 +190,41 @@ def main():
         extr = Extractor()
         if args.run == ['all']:
             rows = extr.db.conn.execute("SELECT proposal, runnr FROM runs").fetchall()
-            print(f"Reprocessing {len(rows)} runs already recorded...")
+
+            # Dictionary of proposal numbers to sets of available runs
+            available_runs = { }
+            # Lists of (proposal, run) tuples
+            runs = []
+            unavailable_runs = []
+
             for proposal, run in rows:
+                if proposal not in available_runs:
+                    available_runs[proposal] = proposal_runs(proposal)
+
+                if run in available_runs[proposal]:
+                    runs.append((proposal, run))
+                else:
+                    unavailable_runs.append((proposal, run))
+
+            print(f"Reprocessing {len(runs)} runs already recorded, skipping {len(unavailable_runs)}...")
+            for proposal, run in runs:
                 extr.extract_and_ingest(proposal, run, match=args.match, mock=args.mock)
         else:
+            available_runs = proposal_runs(extr.db.metameta["proposal"])
+
             try:
-                runs = [int(r) for r in args.run]
+                runs = set([int(r) for r in args.run])
             except ValueError as e:
                 sys.exit(f"Run numbers must be integers ({e})")
-            for run in runs:
+
+            unavailable_runs = runs - available_runs
+            if len(unavailable_runs) > 0:
+                # Note that we print unavailable_runs as a list so it's enclosed
+                # in [] brackets, which is more recognizable than the {} braces
+                # that sets are enclosed in.
+                print(f"Warning: skipping {len(unavailable_runs)} runs because they don't exist: {list(unavailable_runs)}")
+
+            for run in (runs & available_runs):
```

Comment:
```suggestion
                print(f"Warning: skipping {len(unavailable_runs)} runs because they don't exist: {sorted(unavailable_runs)}")

            for run in sorted(runs & available_runs):
```

Otherwise weird things can happen with the ordering:

```
>>> set(range(120, 140))
{128, 129, 130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 120, 121, 122, 123, 124, 125, 126, 127}
```